controllers/autohealconfig_controller.go

```diff
@@ -47,16 +66,410 @@ type AutohealConfigReconciler struct {
+		}
+	}
+
+	if autoheal.DeletionTimestamp == nil {
+		if autoheal.Status.LastSyncPeriod != 0 {
+			lastSync := time.Unix(autoheal.Status.LastSyncPeriod, 0)
+			period, err := time.ParseDuration(autoheal.Spec.SyncPeriod)
+			if autoheal.Spec.SyncPeriod == "" {
+				log.Info("Please specify a sync period")
+			}
+			if err != nil {
+				log.Errorf("Error parsing operator resync period %s", err)
+				return ctrl.Result{}, err
+
+			}
+			nextSync := lastSync.Add(period)
+			if time.Now().Before(nextSync) {
+				return ctrl.Result{
+					Requeue:      true,
+					RequeueAfter: RequeueDelaySuccess,
+				}, nil
+
+			}
+		}
+		configRepo, err := r.GetConfigRepoSecret(ctx, autoheal, &autoheal.Status)
+		if err != nil {
+			log.Errorf("Unable to reconcile getting config repo secret %s", err)
+		}
+
+		if configRepo != nil {
+			files, err := ReadFiles(configRepo)
+			if err != nil {
+				log.Errorf("Unable to get file endpoints from config repo %s", err)
+			}
+			scenarios, err := r.getScenarios(files, configRepo, autoheal, ctx)
+			if err != nil {
+				log.Errorf("Unable to get scenerios from config repo %s", err)
 
-	// TODO(user): your logic here
+			}
+			err = r.createConfigMaps(autoheal, scenarios, ctx)
+			if err != nil {
+				log.Errorf("Unable to create configmaps %s", err)
+			}
+			err = r.removeUnwantedConfigmap(ctx, &scenarios)
+			if err != nil {
+				log.Error("Unable to delete unwanted configmaps")
+			}
+		}
 
-	return ctrl.Result{}, nil
+	} else {
+		err = r.deleteConfigmap(autoheal, ctx)
+		if err != nil {
+			log.Errorf("Unable to reconcile deleting config maps %s", err)
+		}
+
+		err = r.deleteConfigSecret(autoheal, ctx)
+		if err != nil {
+			log.Errorf("Unable to reconcile deleting config secret %s", err)
+		}
+		if autoheal.Finalizers != nil {
+			autoheal.Finalizers = []string{}
+			err = r.Update(ctx, autoheal)
+			if err != nil {
+				log.Errorf("Unable to update finalizer to remove it %s", err)
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	return r.updateStatus(autoheal)
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *AutohealConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&autohealv1alpha1.AutohealConfig{}).
+		For(&v1alpha1.AutohealConfig{}).
 		Complete(r)
 }
+
+// GetConfigRepoSecret Gets the secret needed to provide credentials to sycn with github repo
+func (r *AutohealConfigReconciler) GetConfigRepoSecret(ctx context.Context, autoheal *v1alpha1.AutohealConfig, status *v1alpha1.AutohealConfigStatus) (*v1alpha1.ConfigRepositoryInfo, error) {
+	log.Info("Stage: Reading config secret")
+	configRepoSecret := &core.Secret{}
+	secretSelector := &client.ObjectKey{
+		Namespace: autoheal.Namespace,
+		Name:      AutohealConfigSecret,
+	}
+
+	err := r.Client.Get(ctx, *secretSelector, configRepoSecret)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Infof("Unable to find config repo secret %s", err)
```

Comment:
```suggestion
			log.Info(err)
```
If the secret isn't found, this is the log line that is currently printed to the console: `Unable to find config repo secret Secret "autoheal-config" not found`. It might be enough to just print the err, as that tells us the problem: `Secret "autoheal-config" not found`.